readthedocs/config/models.py

```diff
@@ -49,6 +49,22 @@ class BuildTool(Base):
     __slots__ = ('version', 'full_version')
 
 
+class BuildJobs(Base):
+
+    __slots__ = (
+        "pre_checkout",
+        "post_checkout",
+        "pre_system_dependencies",
+        "post_system_dependencies",
+        "pre_create_environment",
+        "post_create_environment",
+        "pre_install",
+        "post_install",
+        "pre_build",
+        "post_build",
```

Comment:
When we allow overwritting the `build` job, we will have an inner command per asset type:

```yaml
build:
  jobs:
    build:
      html:
        - command 1
        -  command 2
      pdf:
        - command 1
```

Does it make sense to have `pre_build.html` and `pre_build.pdf` as well?

Reply:
cc @stsewd @ericholscher